nullaway/src/test/resources/com/uber/nullaway/testdata/CheckContractNegativeCases.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2017 Uber Technologies, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.nullaway.testdata;
+
+import javax.annotation.Nullable;
+import org.jetbrains.annotations.Contract;
+
+public class CheckContractNegativeCases {
+
+  @Contract("_, !null -> !null")
+  @Nullable
+  Object foo(Object a, @Nullable Object b) {
+    if (a != null) {
```

Comment:
Should this be checking `b != null`?  If this is passing without error right now, looks like something is broken?

Reply:
`b` is not null according to the contract premises. Hence all the returns are are non-null and hence this is safe. Putting `b != null`, will not change anything as our analysis for contract does not depend on conditionals.

Reply:
`b` is not null according to the contract premises. Hence all the returns are are non-null and hence this is safe. Putting `b != null`, will not change anything as our analysis for contract does not depend on conditionals.

Reply:
On another read, I think this test case is ok.  Sorry for the confusion.